### PR TITLE
Add a GitHub workflow definition

### PR DESCRIPTION
**Pull request by dscho**
With this PR, it should be a lot easier to build new installers. An example build can be found [here](https://github.com/dscho/git_osx_installer/actions/runs/799382331).

Please note that this PR only builds unsigned installers for now. If this experiment goes well, we should probably look into integrating both a signing step as well as a step to upload the result to SourceForge ([using their API](https://sourceforge.net/p/forge/documentation/Using%20the%20Release%20API/)), both contingent on the respective credentials being available as [GitHub Action secrets](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets).

### CODE REVIEW

**Comment by dscho** [comment]
Side note for lurkers: to build the installer for the latest Git, just uncomment the `push:` line in `.github/workflows/osx-installer.yml`, commit and push the result to your fork.

**Comment by dscho** [comment]
@timcharper gentle ping?

**Comment by dscho** [comment]
@timcharper another gentle ping? Would be nice to get this going in time for Git v2.27.0...

**Comment by SyntevoAlex** [comment]
@dscho thanks for your work, I'm loving this new GitHub Action!

**Comment by dscho** [comment]
@timcharper hi there! Long time no hear. Any news from your side?

**Review by sokarax — APPROVED** [review]

**Review by bl-ue — APPROVED** [review]
Hey, @timcharper sorry if this is bothering you but do you mind taking a look at this PR?  (@dscho is the maintainer for git-for-windows in case you're not aware).

File: .github/workflows/osx-installer.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  GIT_REPOSITORY: "${{github.event.inputs.git-repository}}"
+  GIT_REF: "${{github.event.inputs.git-ref}}"
+  FALLBACK_GIT_REPOSITORY: https://github.com/git/git
+  FALLBACK_GIT_REF: v2.31.1
+
+  # `gettext` is keg-only
+  LDFLAGS: -L/usr/local/opt/gettext/lib
+  CFLAGS: -I/usr/local/opt/gettext/include
+  # Link with cURL
+  CURL_LDFLAGS: -lcurl
```
Comment (SyntevoAlex) [inline .github/workflows/osx-installer.yml:23]: I wasn't able to figure why is this needed.

According to git's Makefile, it's only needed to override default flags [1]. I tested on macOS 11, 10.15, 10.14 - and `curl-config --libs` always returns `-lcurl`. I also tested on freshly installed macOS 11, and it also returns `-lcurl` there.

[1] https://github.com/git/git/blob/master/Makefile#L53

File: .github/workflows/osx-installer.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  # To make use of the catalogs...
```
Comment (SyntevoAlex) [inline .github/workflows/osx-installer.yml:24]: I suggest a more detailed wording:

Catalogs are required to build man. These are installed through `docbook` [1] and `docbook-xsl` [2] dependencies of `asciidoc` in homebrew.

[1] https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/asciidoc.rb#L27
[2] https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/asciidoc.rb#L26

File: .github/workflows/osx-installer.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          brew link --force gettext
+      - name: Clone git
+        run: |
+          git clone -n "${GIT_REPOSITORY:-$FALLBACK_GIT_REPOSITORY}" git &&
+          git -C git pull origin "${GIT_REF:-$FALLBACK_GIT_REF}"
+      - name: Build GIT-VERSION-FILE and .tar.gz files
+        run: |
+          set -x
+          PATH=/usr/local/bin:$PATH \
+          make -C git -j$(sysctl -n hw.physicalcpu) GIT-VERSION-FILE dist dist-doc
```
Comment (SyntevoAlex) [inline .github/workflows/osx-installer.yml:58]: Is it really necessary to build entire git with `dist`? To my understanding, the goal of this step is to collect built `man`.

**Review by SyntevoAlex — COMMENTED** [review]
Again, thanks for this GitHub action! It is really useful.

I was making another git builder for macOS and noticed a few things in your script.

**Comment by narutoDev** [comment]
I couldn't get this Github Action to work for me. Running it failed in the "Bundle .dmg" task with the error:
`'kFSEventStreamEventFlagItemCloned' has been marked as being introduced in macOS 10.13 here, but the deployment target is macOS 10.6.0`

Weirdly enough I was running the action on macos-latest wich is MacOS 12 so I really don't understand why this issue occurs. Any help would be greatly appreciated. Attached are the logs to the failed build runs on my fork.
Original Action: https://ghostbin.me/63c9c10209aec
After some attempted fixes: https://ghostbin.me/63c9c165318ec

File: .github/workflows/osx-installer.yml (hunk 1 above)
Comment (dscho) [reply .github/workflows/osx-installer.yml:23]: Unfortunately, I do not recall the exact reason why this was needed. I _guess_ at the time the `curl-config` script did not quite work as intended.

File: .github/workflows/osx-installer.yml (hunk 3 above)
Comment (dscho) [reply .github/workflows/osx-installer.yml:58]: Not only `man`, but the way `git_osx_installer` is set up, it does not accept Git's source code in the form of a Git repository, but only in the form of a `.tar` file.

**Comment by dscho** [comment]
> `'kFSEventStreamEventFlagItemCloned' has been marked as being introduced in macOS 10.13 here, but the deployment target is macOS 10.6.0`

This error probably means that you would have to adjust https://github.com/timcharper/git_osx_installer/pull/156/files#diff-de90736762a5cf22e0cbc7531a25565f2b3a90c2e741616362a4c8d4bc4fa950R29 a bit.

As it stands, however, I cannot justify to keep this Pull Request open. There has been exactly 0 feedback from the maintainer of this repository, and my intention was to leverage the maintainer's expertise to avoid duplicating efforts in `microsoft/git`, where a macOS installer is built, too. Due to the lack of interest, `microsoft/git` [has switched away from `git_osx_installer` in the meantime](https://github.com/microsoft/git/pull/514). This concludes my involvement in this repository.
